src/main/java/athleticli/ui/Parser.java

```diff
@@ -338,4 +353,178 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
         return new EditSleepCommand(index, startTime, endTime);
     }
 
+    /**
+     * Parses the raw user input for a diet and returns the corresponding diet object.
+     *
+     * @param commandArgs The raw user input containing the arguments.
+     * @return An object representing the diet.
+     * @throws AthletiException
+     */
+    public static Diet parseDiet(String commandArgs) throws AthletiException {
+        final String caloriesMarkerConstant = "calories/";
```

Comment:
Good use of String constants, but might be preferable to factor the String markers to a separate class to allow for easier changes to be made

Reply:
Good idea. We can open an issue for it.